backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java

```diff
@@ -60,7 +60,21 @@ public void getExperimentRunsInExperiment(
   public void getExperimentRunById(
       GetExperimentRunById request,
       StreamObserver<GetExperimentRunById.Response> responseObserver) {
-    super.getExperimentRunById(request, responseObserver);
+    try {
+      final var response =
+          futureExperimentRunDAO
+              .findExperimentRuns(
+                  FindExperimentRuns.newBuilder().addExperimentRunIds(request.getId()).build())
+              .thenApply(
+                  findResponse ->
+                      GetExperimentRunById.Response.newBuilder()
+                          .setExperimentRun(findResponse.getExperimentRuns(0))
```

Comment:
You have no guarantee you'll have at least one result in the response.

Reply:
Throw error if more than one run found for given id

Reply:
This is exactly the opposite of what I said. I said that we might not even have one result, as the `get` assumes there is at least one element. I'm not concerned about the case of more than one result.

Reply:
We correctly handle the case of none found in https://github.com/VertaAI/modeldb/pull/2239/files

Reply:
Handled 404 case as well